projects/components/src/download-file/download-file.component.ts

```diff
@@ -36,9 +37,12 @@ export class DownloadFileComponent {
 
   public triggerDownload(metadata: DownloadFileMetadata): void {
     this.dataLoading = true;
-    this.fileDownloadService.downloadAsText(metadata).subscribe(() => {
-      this.dataLoading = false;
-      this.cdr.detectChanges();
-    });
+    this.fileDownloadService
+      .downloadAsText(metadata)
+      .pipe(take(1))
```

Comment:
Finite observable; upstream emits once.

Reply:
do you want me to revert this or any other similar change in this PR if it is not breaking anything?

Reply:
Just to capture what I was saying offline - I think the readability is key. Making double sure a subscription is cleaned up is fine as long as we're not making the code significantly harder to follow. I'd put `take` in the former camp, but subscription lifecycle (in its current form) in the latter.

Reply:
Just my 2c though, up to you guys.

Reply:
Okay. let me update